### PR TITLE
[WFLY-12252]: Add metric type flags on messaging subsytem metrics.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding the missing flags

Jira: https://issues.jboss.org/browse/WFLY-12252

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java [hunk 1]
```diff
@@ -100,6 +100,7 @@
     AttributeDefinition CONSUMER_COUNT = create("consumer-count", INT)
             .setStorageRuntime()
             .setRequired(false)
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java]: it's a gauge (number of consumers can go up and down) as a counter is only monotonically increasing

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java [hunk 2]
```diff
@@ -131,6 +132,7 @@
 
     AttributeDefinition DELIVERING_COUNT = create("delivering-count", INT)
             .setStorageRuntime()
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java [hunk 3]
```diff
@@ -230,12 +232,13 @@
     AttributeDefinition MESSAGE_COUNT = create("message-count", LONG)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java [hunk 4]
```diff
@@ -300,6 +303,7 @@
     AttributeDefinition SCHEDULED_COUNT = create("scheduled-count", LONG)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java [hunk 5]
```diff
@@ -75,26 +76,31 @@
     static final AttributeDefinition DURABLE_MESSAGE_COUNT = create(CommonAttributes.DURABLE_MESSAGE_COUNT, INT)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java [hunk 6]
```diff
@@ -75,26 +76,31 @@
...
             .build();
 
     static final AttributeDefinition NON_DURABLE_MESSAGE_COUNT = create(CommonAttributes.NON_DURABLE_MESSAGE_COUNT, INT)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java [hunk 7]
```diff
@@ -75,26 +76,31 @@
...
             .build();
 
     static final AttributeDefinition SUBSCRIPTION_COUNT = create(CommonAttributes.SUBSCRIPTION_COUNT, INT)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java [hunk 8]
```diff
@@ -75,26 +76,31 @@
...
             .build();
 
     static final AttributeDefinition DURABLE_SUBSCRIPTION_COUNT = create(CommonAttributes.DURABLE_SUBSCRIPTION_COUNT, INT)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java]: gauge

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java [hunk 9]
```diff
@@ -75,26 +76,31 @@
...
             .build();
 
     static final AttributeDefinition NON_DURABLE_SUBSCRIPTION_COUNT = create(CommonAttributes.NON_DURABLE_SUBSCRIPTION_COUNT, INT)
             .setStorageRuntime()
             .setUndefinedMetricValue(new ModelNode(0))
+            .addFlag(COUNTER_METRIC)
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicDefinition.java]: gauge

**Review by jmesnil — CHANGES_REQUESTED** [review]

**Comment by ehsavoie** [comment]
All resolved

**Comment by bstansberry** [comment]
@jmesnil OK now?

**Review by jmesnil — APPROVED** [review]

**Pull request merged**